packages/babel-types/src/clone/cloneNode.ts

```diff
@@ -32,6 +33,9 @@ export default function cloneNode<T extends t.Node>(
 ): T {
   if (!node) return node;
 
+  const isTop = !commentsCache.get("inCloning");
+  if (isTop) commentsCache.set("inCloning", true);
```

Comment:
Instead of a special string key, we can just default `commentsCache` to `null` and do
```js
const isTop = !commentsCache;
if (isTop) commentsCache = new Map();
```
and replace `commentsCache.clear()`with `commentsCache = null`.

Reply:
Or well actually, even better we can avoid relying on a global map:

```js
export default function cloneNode<T extends t.Node>(
  node: T,
  deep: boolean = true,
  withoutLoc: boolean = false,
): T {
  return cloneNodeInternal(node, deep, withoutLoc, new Map());
}

function cloneNodeInternal<T extends t.Node>(
  node: T,
  deep: boolean,
  withoutLoc: boolean,
  commentsCache: Map<t.Comment, t.Comment>
): T {
  // original cloneNode implementation, recursively passing `commentsCache` to nested `cloneNodeInternal` calls
}
```